src/driver/mod.rs

```diff
@@ -145,10 +145,39 @@ impl Drop for Driver {
 
         // pre-determine what to cancel
         let mut cancellable_ops = Vec::new();
-        for (id, cycle) in self.ops.lifecycle.iter() {
-            // don't cancel completed items
-            if !matches!(cycle, Lifecycle::Completed(_)) {
-                cancellable_ops.push(id);
+
+        let Ops {
+            lifecycle,
+            completions,
+        } = &mut self.ops;
+
+        for (id, cycle) in lifecycle.iter_mut() {
```

Comment:
Just like the code it is replacing, this will iterate through the full length of the Vec that has been allocated for the slab, even if the slab could report a 0 length to us. Also I think the slab, when it is dropped, will be drop the vec normally, so that will also be iterating through the entire Vec, looking for Some(T) entries that need to have their drop called. Guess we shouldn't try to change the slab drop, so the Vec will be brought into cache one way or the other anyway.